universal-application-tool-0.0.1/app/controllers/applicant/ApplicantProgramBlocksController.java

```diff
@@ -70,7 +72,18 @@ public ApplicantProgramBlocksController(
                 return notFound();
               }
             },
-            httpExecutionContext.current());
+            httpExecutionContext.current())
+        .exceptionally(
+            ex -> {
```

Comment:
(optional): should this be refactored into a method that looks like (loosely)

```
public Result handleExceptions(ImmutableSet<Class<? extends Exception>> okExceptionClasses, Throwable exception) {
  for (Class okExceptionClass : okExceptionClasses) {
    if (okExceptionClass.isInstance(exception)) {
      return notFound(exception.toString());    
    }
  }
 throw new RuntimeException(exception);
}
```

My method and arg names are not great.

I can imagine some helper methods to construct the `ImmutableSet` from the bottom up, so we don't have to manage things so manually.

Reply:
the `if` case probably log the error and return `notFound("Unable to process this request.")`

Reply:
That sounds like a great idea. Opened #487 and would like to merge this PR first.